x-pack/plugins/index_management/server/routes/api/inference_models/register_create_elasticsearch_inference_route.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { addBasePath } from '..';
+import { RouteDependencies } from '../../../types';
+
+export function registerCreateElasticsearchInferenceRoute({
+  router,
+  lib: { handleEsError },
+  config,
+}: RouteDependencies) {
+  const paramsSchema = schema.object({
+    inferenceId: schema.string(),
+  });
+  const bodySchema = schema.object({
+    trainedModelId: schema.string(),
+  });
+  // Create text_embedding inference
+  router.post(
+    {
+      path: addBasePath('/inference/text_embedding/{inferenceId}'),
```

Comment:
Could you please add api integration tests for the new route? For an example, see this folder `x-pack/test/api_integration/apis/management/index_management`.
Also, if the route is not intended for public use, I would recommend using an internal route with `addInternalBasePath` instead of `addBasePath`.

Reply:
Right now, we already have a route to create inference_endpoint. I remove this code and utilized the existing one.